processor/spanmetricsprocessor/processor.go

```diff
@@ -259,31 +288,21 @@ func (p *processorImp) tracesToMetrics(ctx context.Context, traces ptrace.Traces
 	// This component no longer needs to read the metrics once built, so it is safe to unlock.
 	p.lock.Unlock()
 
-	if err != nil {
-		return err
-	}
-
-	if err = p.metricsExporter.ConsumeMetrics(ctx, m); err != nil {
-		return err
+	if err := p.metricsExporter.ConsumeMetrics(ctx, m); err != nil {
+		p.logger.Error("Failed ConsumeMetrics: " + err.Error())
```

Comment:
I considered accumulating errors and returning them on the next trace batch received, but since those errors no longer relate to that batch of traces, I felt it didn't make sense to couple those errors. Ideas are very welcome. 😄